In [1]:
import pandas as pd
features_df = pd.read_csv("/home/jbetancur/Desktop/codes/clustering/feature_extraction/output/wavelet_pyradiomic_aggregated_lesion_features.csv")
patients_df = pd.read_excel("/home/jbetancur/Desktop/codes/clustering/data/Metadata/25-11-2025_xnatt_brain.xlsx")


/tmp/ipykernel_1662093/1972457494.py:2: DtypeWarning: Columns (647) have mixed types. Specify dtype option on import or set low_memory=False.
  features_df = pd.read_csv("/home/jbetancur/Desktop/codes/clustering/feature_extraction/output/wavelet_pyradiomic_aggregated_lesion_features.csv")


In [10]:
features_df.head()

label_id  num_voxels  volume_physical  T1_3d_original_shape_Elongation  \
0         1          17        16.999091                         0.956726   
1         2           2         1.999893                              NaN   
2         3          12        11.999358                         0.612405   
3         4          45        44.997593                         0.557967   
4         5           4         3.999786                         0.999947   

   T1_3d_original_shape_Flatness  T1_3d_original_shape_LeastAxisLength  \
0                       0.591200                              1.986722   
1                            NaN                                   NaN   
2                       0.612405                              2.000000   
3                       0.000000                              0.000000   
4                       0.000000                              0.000000   

   T1_3d_original_shape_MajorAxisLength  \
0                              3.360493   
1                                   NaN   
2                              3.265812   
3                             10.303593   
4                              2.000000   

   T1_3d_original_shape_Maximum2DDiameterColumn  \
0                                      3.162125   
1                                           NaN   
2                                      3.162125   
3                                      5.999679   
4                                      1.999893   

   T1_3d_original_shape_Maximum2DDiameterRow  \
0                                   3.162278   
1                                        NaN   
2                                   2.236068   
3                                   9.000000   
4                                   2.000000   

   T1_3d_original_shape_Maximum2DDiameterSlice  ...  \
0                                     3.605492  ...   
1                                          NaN  ...   
2                                     3.162125  ...   
3                                    10.295500  ...   
4                                     2.236044  ...   

   QSM_2d_original_ngtdm_Strength  PatientID  FolderType  YearFolder  \
0                        0.912548   14109676    baseline        2022   
1                             NaN   14109676    baseline        2022   
2                        1.153846   14109676    baseline        2022   
3                        0.145467   14109676    baseline        2022   
4                        1.000000   14109676    baseline        2022   

                                         T1_3d_error  \
0                                                NaN   
1  mask has too few dimensions (number of dimensi...   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                         T1_2d_error  \
0                                                NaN   
1  mask has too few dimensions (number of dimensi...   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                         T2_3d_error  \
0                                                NaN   
1  mask has too few dimensions (number of dimensi...   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                         T2_2d_error  \
0                                                NaN   
1  mask has too few dimensions (number of dimensi...   
2                                                NaN   
3                                                NaN   
4                                                NaN   

                                        QSM_3d_error  \
0                                                NaN   
1  mask has too 

In [8]:
type(patients_df['patient'][0])

numpy.int64

In [3]:
patients_only = patients_df[patients_df['patient'] == 1]
controls_only = patients_df[patients_df['patient'] == 0]

print(patients_only.shape)
print(controls_only.shape)

(87, 471)
(44, 471)


In [9]:
feature_patients = features_df[features_df['PatientID'].isin(patients_only['subject_id'])]
feature_controls = features_df[features_df['PatientID'].isin(controls_only['subject_id'])]

print(feature_patients.shape)
print(feature_controls.shape)

(2786, 654)
(848, 654)


In [12]:
baseline_count = feature_patients[feature_patients['FolderType'].str.lower() == 'baseline']['PatientID'].nunique()
followup_count = feature_patients[feature_patients['FolderType'].str.lower() == 'follow_up']['PatientID'].nunique()

print("Unique patient count in baseline: ", baseline_count)
print("Unique patient count in followup: ", followup_count)

Unique patient count in baseline:  40
Unique patient count in followup:  32


In [15]:
baseline_count_controls = feature_controls[feature_controls['FolderType'].str.lower() == 'baseline']['PatientID'].nunique()
followup_count_controls = feature_controls[feature_controls['FolderType'].str.lower() == 'follow_up']['PatientID'].nunique()

print("Unique control count in baseline:", baseline_count_controls)
print("Unique control count in follow up:", followup_count_controls)

Unique control count in baseline: 23
Unique control count in follow up: 20


In [17]:
baseline_ids = feature_patients[feature_patients['FolderType'].str.lower() == 'baseline']['PatientID'].unique().tolist()
followup_ids = feature_patients[feature_patients['FolderType'].str.lower() == 'follow_up']['PatientID'].unique().tolist()
only_baseline = set(baseline_ids) - set(followup_ids)
only_followup = set(followup_ids) - set(baseline_ids)

print("Patients only at baseline: ", only_baseline)
print("Patients only at followup: ", only_followup)

Patients only at baseline:  {12521539, 13146694, 14543433, 14370736, 12624080, 10847986, 11974835, 16884118, 12591126, 16836063}
Patients only at followup:  {10428634, 10913259}


- **10847986** - only has baseline
- **11974835** - only has baseline
- **16884118** - only has baseline

12521539 -> empty mask, error itksnap.
12591126 -> It actually has the masks in the correct format. We would need to re register the mask. SHould be easy with the transformation matrix file.
12624080 -> Nothing from XNAT.
13146694 -> Nothing from XNAT.
14370736 -> Nothing from XNAT.
14543433 -> Nothing from XNAT.
16836063 -> Nothing from XNAT.